src/commands/cmd_string.cc

```diff
@@ -620,6 +621,201 @@ class CommandCAD : public Commander {
   }
 };
 
+class CommandLCS : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    CommandParser parser(args, 3);
+    while (parser.Good()) {
+      if (parser.EatEqICase("IDX")) {
+        get_idx_ = true;
+      } else if (parser.EatEqICase("LEN")) {
+        get_len_ = true;
+      } else if (parser.EatEqICase("WITHMATCHLEN")) {
+        with_match_len_ = true;
+      } else if (parser.EatEqICase("MINMATCHLEN")) {
+        min_match_len_ = GET_OR_RET(parser.TakeInt<int64_t>());
+        if (min_match_len_ < 0) {
+          min_match_len_ = 0;
+        }
+      } else {
+        return parser.InvalidSyntax();
+      }
+    }
+
+    // Complain if the user passed ambiguous parameters.
+    if (get_idx_ && get_len_) {
+      return {Status::RedisParseErr,
+              "If you want both the length and indexes, "
+              "please just use IDX."};
+    }
+
+    return Status::OK();
+  }
+
+  Status Execute(Server *srv, Connection *conn, std::string *output) override {
+    redis::String string_db(srv->storage, conn->GetNamespace());
+
+    std::string a;
+    std::string b;
+    auto s1 = string_db.Get(args_[1], &a);
+    auto s2 = string_db.Get(args_[2], &b);
+
+    if (!s1.ok() && !s1.IsNotFound()) {
+      return {Status::RedisExecErr, s1.ToString()};
+    }
+    if (!s2.ok() && !s2.IsNotFound()) {
+      return {Status::RedisExecErr, s2.ToString()};
+    }
+    if (s1.IsNotFound()) a = "";
+    if (s2.IsNotFound()) b = "";
+
+    // Detect string truncation or later overflows.
+    if (a.length() >= UINT32_MAX - 1 || b.length() >= UINT32_MAX - 1) {
+      return {Status::RedisExecErr, "String too long for LCS"};
+    }
+
+    // Compute the LCS using the vanilla dynamic programming technique of
+    // building a table of LCS(x, y) substrings.
+    auto alen = static_cast<uint32_t>(a.length());
+    auto blen = static_cast<uint32_t>(b.length());
+
+    // Allocate the LCS table.
+    uint64_t dp_size = (alen + 1) * (blen + 1);
+    uint64_t bulk_size = dp_size * sizeof(uint32_t);
```

Comment:
Should we better check `dp_size` not overflow uint32_t? Since the memory usage will be 16GB if dp_size reaches `numeric_limits<uint32_t>::max()` (Though redis uses same algorithm)